configuration/secrets.rst

```diff
@@ -0,0 +1,319 @@
+            <container xmlns="http://symfony.com/schema/dic/services"
+                xmlns:framework="http://symfony.com/schema/dic/framework"
+                xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                xsi:schemaLocation="http://symfony.com/schema/dic/services https://symfony.com/schema/dic/services/services-1.0.xsd
+                    http://symfony.com/schema/dic/framework https://symfony.com/schema/dic/framework/framework-1.0.xsd"
+            >
+                <framework:config secret="%env(APP_SECRET)%">
+                    <framework:secrets/>
+                </framework:config>
+            </container>
+
+    .. code-block:: php
+
+        // config/packages/framework.php
+        $container->loadFromExtension('framework', [
+            'secrets' => true,
+        ]);
+
+.. tip::
+
+    The ``encryption_key`` configuration parameter accept both a path to an
+    encryption key or the content of the key itself. It allows you, for
+    instance, to store the content of the key in an environment variable and
+    refer it with ``%env(PRIVATE_KEY)%``.
+
+.. _secrets-generate-key:
+
+Generate an Encryption Key
+--------------------------
+
+Before creating a new ``secret``, you need to create an ``encryption key``.
+This can be done with the provided command ``secrets:generate-key``.
+
+.. code-block:: terminal
+
+    $ APP_ENV=prod php bin/console secrets:generate-key
+
+This command generate a new ``encryption key`` in
+``%kernel.project_dir%/config/secrets/encryption_%kernel.environment%.key``
+
+.. note::
+
+    In order to use Symfony's built-in secret storage, you will need the
+    `libsodium`_ PHP extension or use the `sodium_compat`_ package.
+
+Symfony generates a key with a symetrics algorithm, meaning that this key could
+be used to both encrypt **and** decrypt secrets. The number of people who
+possess this key should be as small as possible.
+
+.. caution::
+
+    This file is sensitive and **must not** be commited nor publicly shared.
+    Every developpers and CI don't need that key. If the encryption key have
+    been exposed (ex-employee leaving for instance), you should consider
+    regenerating a new one.
+
+.. _secrets-add:
+
+Create a Secret
+---------------
+
+Once the ``encryption key`` generated, you can add new secret with the command
+``secrets:add``. Symfony will ask you to enter the text to encrypt and generate
+a new file containing the ciphered text in a file stored by default in the
+folder ``%kernel.project_dir%/config/secrets/%kernel.environment%/``. This file
+should be commited along side the other project's files.
+
+.. code-block:: terminal
+
+    $ APP_ENV=prod php bin/console secrets:add DATABASE_PASSWORD
+
+.. tip::
+
+    If the ``encryption key`` is compromized, you can regenerate a new key with
+    the command ``secrets:generate-key``. Symfony will decrypt the previous
+    secret with the old key, and re-encrypt theme with the new one.
```

Comment:
decrypt previous secrets

If you use them after (no e at the end)

Reply:
fixed